content/open-source/sigstore/cosign/how-to-sign-an-sbom-with-cosign.md

```diff
@@ -17,143 +17,270 @@ toc: true
+[...]
+```
 
-## Sign the SBOM
+If we choose, we can further parse this output so we can examine the `cowsay` package we added in our `Dockerfile`:
+
+```sh
+cosign download attestation \
+ $DIGEST | \
+ jq -r .payload | base64 -d |\
+ jq .predicate | jq .packages | \
+ jq '.[] | select(.name == "cowsay")'
+```
+
+```json
+{
+  "SPDXID": "SPDXRef-Package-apk-cowsay-f6f3edc220b6705a",
+  "copyrightText": "NOASSERTION",
+  "description": "Configurable talking cow (and a few other creatures)",
+  "downloadLocation": "NONE",
+  "externalRefs": [
+    {
+      "referenceCategory": "SECURITY",
+      "referenceLocator": "cpe:2.3:a:cowsay:cowsay:3.04-r0:*:*:*:*:*:*:*",
+      "referenceType": "cpe23Type"
+    },
+    {
+      "referenceCategory": "PACKAGE-MANAGER",
+      "referenceLocator": "pkg:apk/wolfi/cowsay@3.04-r0?arch=x86_64&distro=wolfi-20230201",
+      "referenceType": "purl"
+    }
+  ],
+  "filesAnalyzed": true,
+  "licenseConcluded": "NOASSERTION",
+  "licenseDeclared": "GPL-2.0-or-later",
+  "name": "cowsay",
+  "packageVerificationCode": {
+    "packageVerificationCodeValue": "4f82bdba8e1217f8af0abee5cadc9c2387bf4720"
+  },
+  "sourceInfo": "acquired package info from APK DB: /lib/apk/db/installed",
+  "supplier": "NOASSERTION",
+  "versionInfo": "3.04-r0"
+}
+```
 
-You will sign the SBOM in a similar way to signing other software artifacts. 
+At this point, we've attested to the contents of our image with an SBOM and confirmed that the attestation is attached to the image in our Docker Hub repository. In the next section, we'll learn how to verify the identity of the entity issuing an attestation.
 
-Make sure you are in the correct local directory for your Cosign key pair. If you generated the key pair in the signed container example, it will be in your home user directory, so make sure you move your present working directory there with the `cd ~` command.
+## Verifying an Attestation
 
-You’ll be signing the SBOM with the SHA that you received in the output from the previous command. This is a long string that starts with `sha256` and ends with `.sbom`. You can verify that this was pushed to the container registry by checking the web user interface of Docker Hub or alternate registry. 
+Cosign can also be used to verify the identity of the person or entity issuing an attestation. The following assumes you used GitHub to authenticate when attesting in the previous step.
 
-We have an example SBOM SHA below, please use the SHA output you received.
+To verify that an attestation was issued by a specific entity, we use the `cosign verify-attestation` command, specifying the email address of the issuer:
 
 ```sh
-cosign sign --key cosign.key docker-username/hello-container:sha256-690ecfd885f008330a66d08be13dc6c115a439e1cc935c04d181d7116e198f9c.sbom
+cosign verify-attestation \
+ --certificate-oidc-issuer=https://github.com/login/oauth \
+ --type https://spdx.dev/Document \
+ --certificate-identity=<emailaddress@emailprovider.com> \
```

Comment:
```suggestion
 --certificate-identity=emailaddress@emailprovider.com \
```

Reply:
this can be confusing 😅 I just spent a long time trying to make this work here and then I realized I needed to take out the `<` `>`